transmogrifier/sources/xml/dspace_dim.py

```diff
@@ -260,47 +183,174 @@ def get_optional_fields(self, xml: Tag) -> dict | None:
         return fields
 
     @classmethod
-    def get_content_types(cls, xml: Tag) -> list[str] | None:
-        """
-        Retrieve content types from a DSpace DIM XML record.
+    def get_alternate_titles(
+        cls, source_record: Tag
+    ) -> list[timdex.AlternateTitle] | None:
+        alternate_titles = [
+            timdex.AlternateTitle(
+                value=str(alternate_title.string),
+                kind=alternate_title["qualifier"],
+            )
+            for alternate_title in source_record.find_all(
+                "dim:field", element="title", string=True
+            )
+            if alternate_title.get("qualifier")
+        ]
+        # If the record has more than one main title, add extras to alternate_titles
+        alternate_titles.extend(
+            [
+                timdex.AlternateTitle(value=title)
+                for title in cls.get_main_titles(source_record)[1:]
+            ]
+        )
+        return alternate_titles or None
 
-        May be overridden by source subclasses that retrieve content type values
-        differently.
+    @classmethod
+    def get_citation(cls, source_record: Tag) -> str | None:
+        if citation := source_record.find(
+            "dim:field", element="identifier", qualifier="citation", string=True
+        ):
+            return citation.string
+        return None
 
-        Args:
-            xml: A BeautifulSoup Tag representing a single DSpace DIM XML record.
-        """
+    @classmethod
+    def get_content_type(cls, source_record: Tag) -> list[str] | None:
         return [
-            t.string for t in xml.find_all("dim:field", element="type", string=True)
+            str(content_type.string)
+            for content_type in source_record.find_all(
+                "dim:field", element="type", string=True
+            )
         ] or None
 
     @classmethod
-    def get_main_titles(cls, xml: Tag) -> list[str]:
+    def get_contents(cls, source_record: Tag) -> list[str] | None:
+        return [
+            contents.string
+            for contents in source_record.find_all(
+                "dim:field",
+                element="description",
+                qualifier="tableofcontents",
+                string=True,
+            )
+        ] or None
+
+    @classmethod
+    def get_contributors(cls, source_record: Tag) -> list[timdex.Contributor] | None:
+        contributors = []
+        contributors.extend(list(cls._get_creators(source_record)))
+        contributors.extend(
+            list(cls._get_contributors_by_contributor_element(source_record))
+        )
```

Comment:
Related to the comment below asking about use of `yield`, I think we could drop the `list(...)` to convert the generator returns from `_get_creators()` and `_get_contributors...()`. 

I had to test to confirm, but it looks like `<list>.extend()` will hydrate the results of the generator if passed (which kind of makes sense if you think about it, as it's modifying the original list with values):

```python
In [3]: def foo():
   ...:     for x in range(0,10):
   ...:         yield x
   ...: 

In [4]: y = []

In [5]: y.extend(foo())

In [6]: y
Out[6]: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
```

Learn something new everyday!  Don't think it hurts to have the `list(...)` here, but just noting.

Reply:
Interesting, I never reconsidered this pattern but glad you pointed it out! I'll remove `list(...)`